src/coreComponents/physicsSolvers/surfaceGeneration/SurfaceGenerator.cpp

```diff
@@ -2893,322 +2907,313 @@ void SurfaceGenerator::calculateNodeAndFaceSif( DomainPartition const & domain,
     displacement.move( hostMemorySpace, false );
   } );
 
-  for( localIndex const trailingFaceIndex : m_trailingFaces )
-//  RAJA::forall< parallelHostPolicy >( RAJA::TypedRangeSegment< localIndex >( 0, m_trailingFaces.size() ),
-//                                      [=] GEOS_HOST_DEVICE ( localIndex const trailingFacesCounter )
+  // auto nodalForceKernel = surfaceGenerationKernels::createKernel( elementManager, constitutiveManager,
+  //                                                                 viewKeyStruct::solidMaterialNameString(), false );
+
+  surfaceGenerationKernels::kernelSelector( elementManager,
+                                            constitutiveManager,
+                                            viewKeyStruct::solidMaterialNameString(),
+                                            m_isPoroelastic, [&] ( auto nodalForceKernel )
   {
-//    localIndex const trailingFaceIndex = m_trailingFaces[ trailingFacesCounter ];
+    for( localIndex const trailingFaceIndex : m_trailingFaces )
+    {
+      //  RAJA::forall< parallelHostPolicy >( RAJA::TypedRangeSegment< localIndex >( 0, m_trailingFaces.size() ), [=] GEOS_HOST_DEVICE (
```

Comment:
same here

Reply:
I don't think these have ever been raja loops tbh. We won't be ale to execute them on device so I don't know if it's worth changing it to a raja loop. Makes the syntax heavier.

Reply:
ok, sounds good